src/mesh/X3D_Draco_Mesh_Reader.cc

```diff
@@ -0,0 +1,234 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   mesh/X3D_Draco_Mesh_Reader.cc
+ * \author Ryan Wollaeger <wollaeger@lanl.gov>, Kendra Keady
+ * \date   Thursday, Jul 12, 2018, 08:46 am
+ * \brief  X3D_Draco_Mesh_Reader class implementation file.
+ * \note   Copyright (C) 2018 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#include "X3D_Draco_Mesh_Reader.hh"
+#include "ds++/DracoStrings.hh"
+#include <fstream>
+#include <iostream>
+
+namespace rtt_mesh {
+
+//---------------------------------------------------------------------------//
+// CONSTRUCTOR
+//---------------------------------------------------------------------------//
+/*!
+ * \brief X3D_Draco_Mesh_Reader constructor.
+ *
+ * \param[in] filename_ name of file to be parsed
+ */
+X3D_Draco_Mesh_Reader::X3D_Draco_Mesh_Reader(std::string filename_)
+    : filename(filename_) {
+  // check for valid file name
+  Insist(filename_.size() > 0, "No file name supplied.");
+  // \todo add more filename criteria?
+}
+
+//---------------------------------------------------------------------------//
+// PUBLIC FUNCTIONS
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Build the cell-face index map to the corresponding coordinates.
+ */
+void X3D_Draco_Mesh_Reader::read_mesh() {
+
+  Require(parsed_pairs.size() == 0);
+
+  // STEP 1: open the file stream
+
+  std::ifstream x3dfile(filename.c_str());
+
+  // file must exist and be readable
+  Check(x3dfile.good());
```

Comment:
Change to `Insist`.

Reply:
This has been changed to an `Insist`, and also made an `is_open` check as for RTT.